testing/integration/kubernetes_agent_standalone_test.go

```diff
@@ -479,6 +480,215 @@ func TestKubernetesAgentHelm(t *testing.T) {
+			// - clusterWide Deployment  (1 agent pod)
+			// - ksmSharded Statefulset  (1 agent pod)
+			atLeastValidatedPodsNumber: 3,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			ctx := context.Background()
+			hasher := sha256.New()
+			hasher.Write([]byte(tc.name))
+			testNamespace := strings.ToLower(base64.URLEncoding.EncodeToString(hasher.Sum(nil)))
+			testNamespace = noSpecialCharsRegexp.ReplaceAllString(testNamespace, "")
+
+			settings := cli.New()
+			settings.SetNamespace(testNamespace)
+			actionConfig := &action.Configuration{}
+
+			helmChart, err := loader.Load(agentK8SHelm)
+			require.NoError(t, err, "failed to load helm chart")
+
+			err = actionConfig.Init(settings.RESTClientGetter(), settings.Namespace(), "",
+				func(format string, v ...interface{}) {})
+			require.NoError(t, err, "failed to init helm action config")
+
+			helmValues := tc.values
+
+			t.Cleanup(func() {
+				if t.Failed() {
+					k8sDumpAllPodLogs(ctx, client, testNamespace, testNamespace, testLogsBasePath)
+				}
+
+				uninstallAction := action.NewUninstall(actionConfig)
+				uninstallAction.Wait = true
+
+				_, err = uninstallAction.Run("helm-agent")
+				if err != nil {
+					require.NoError(t, err, "failed to uninstall helm chart")
+				}
+			})
+
+			installAction := action.NewInstall(actionConfig)
+			installAction.Namespace = testNamespace
+			installAction.CreateNamespace = true
+			installAction.UseReleaseName = true
+			installAction.ReleaseName = "helm-agent"
+			installAction.Timeout = 2 * time.Minute
+			installAction.Wait = true
+			installAction.WaitForJobs = true
+			_, err = installAction.Run(helmChart, helmValues)
+			require.NoError(t, err, "failed to install helm chart")
+
+			podList := &corev1.PodList{}
+			err = client.Resources(testNamespace).List(ctx, podList)
+			require.NoError(t, err, fmt.Sprintf("failed to list pods in namespace %s", testNamespace))
+
+			checkedAgentContainers := 0
+
+			for _, pod := range podList.Items {
+				if !strings.HasPrefix(pod.GetName(), "agent-") {
+					continue
+				}
+
+				command := []string{"elastic-agent", "status"}
+				var stdout, stderr bytes.Buffer
+				var agentHealthyErr error
+				// we will wait maximum 120 seconds for the agent to report healthy
+				for i := 0; i < 120; i++ {
+					stdout.Reset()
+					stderr.Reset()
+					agentHealthyErr = client.Resources().ExecInPod(ctx, testNamespace, pod.Name, "agent", command, &stdout, &stderr)
+					if agentHealthyErr == nil {
+						break
+					}
+					time.Sleep(time.Second * 1)
+				}
+
+				statusString := stdout.String()
+				if agentHealthyErr != nil {
+					t.Errorf("elastic-agent never reported healthy: %v", agentHealthyErr)
+					t.Logf("stdout: %s\n", statusString)
+					t.Logf("stderr: %s\n", stderr.String())
+					t.FailNow()
+					return
+				}
+
+				stdout.Reset()
+				stderr.Reset()
+
+				upgradeCmd := []string{"elastic-agent", "upgrade", "1.0.0"}
```

Comment:
sorry, but is it too much to ask to align this test with the new structure of k8s integration tests? e.g. define a k8sTestStep that checks that the upgrade cmd fails, and just add this step in one of the existing k8s tests?! 🙂

Reply:
Looking into it right away